docs/pairing-architecture.md

```diff
@@ -0,0 +1,284 @@
+
+```
+{
+  message: "pair:auth:metadata",
+  data: {
+    email,
+    avatar,
+    displayName,
+    deviceName,
+  },
+}
+```
+
+
+#### pair:auth:authorize
+
+Direction: Auth -> Supp
+
+Description: Sent by the Auth once the pairing has been confirmed by the user on this side. Contains an OAuth code that can be traded for a token.
+
+```
+{
+  message: "pair:auth:authorize",
+  data: {
+    // See the OAuth server API doc for a description of the parameters.
+    code,
+    state,
+    redirect,
+  },
+}
+```
+
+#### pair:supp:authorize
+
+Direction: Supp -> Auth
+
+Description: Sent by the Supp once the pairing has been confirmed by the user on this side.
+
+```
+{
+  message: "pair:supp:authorize",
+}
+```
+
+
+### Desktop WebChannel messages
+
+On Desktop, the Firefox Accounts Web Content page communicates with the native code (which owns the pairing channel connection) using WebChannel.
+WebChannel messages can only be initiated from the content side, however the native side can send a response.
+
+#### fxaccounts:pair_supplicant_metadata
+
+Description: Allows the web content to request the supp’s device information.
+
+Expected response:
+```
+{
+  ua,
+  city,
+  region,
+  country,
+  ipAddress,
+}
+```
+
+#### fxaccounts:pair_authorize
+
+Allows the web content to inform the native side the confirmation dialog has been accepted.
+
+Expected response: N/A
+
+#### fxaccounts:pair_heartbeat
+
+Description: Sent by the web content every second or so. Allows to surface errors but to also transition to a "completed" screen once the supp side has confirmed the pairing request.
+
+Expected Response: 
+```
+{
+  err, // String
+}
+<OR>
+{
+  suppAuthorized, // Boolean
+}
+
+```
+
+#### fxaccounts:pair_complete
+
+Description: Allows the web content to inform the native side it is aware the flow is finished, has shown a confirmation screen and the resources can be finalized on the native side.
+
+Expected Response: N/A
+
+
+## Other Details
+
+### Routes
+
+* The Mobile (Supplicant) side, implements the following URL: https://accounts.firefox.com/pair/supp
+
+This view accepts OAuth query parameters equivalent to those on the existing /oauth/ view, and will accept the channel id and channel key as parameters in the URL hash fragment. 
+The view performs the entire Mobile side of the protocol, including:
+ * connecting to the provided channel via websocket
+ * passing the OAuth request to the Desktop over the websocket channel
+ * displaying a confirmation screen using the account metadata information sent by Desktop
+ * receiving OAuth code and state from Desktop over the websocket channel.
+ * completing the OAuth flow by redirecting back to the relier
+
+
+* For the Desktop side, we will implement a new view at the following URL: https://accounts.firefox.com/pair/auth
+
+This view is responsible for performing the entire Desktop side of the protocol, including:
```

Comment:
This doesn't seem right, it suggests that the web-content view is responsible for all these things, while in fact many of them are handled in desktop native code.

Reply:
I wonder if these steps are from the previous prototype which was implemented entirely in web content.